src/printer.js

```diff
@@ -2843,7 +2843,11 @@ function printBinaryishExpressions(path, parts, print, options, isNested) {
     // the other ones since we don't call the normal print on BinaryExpression,
     // only for the left and right parts
     if (isNested && node.comments) {
-      parts.push(comments.printComments(path, p => "", options));
+      parts.splice(
+        0,
+        parts.length,
+        comments.printComments(path, p => concat(parts.slice()), options)
+      );
```

Comment:
Where else is the instance being mutated that would create an inconsistency? If you reassign the variable any mutations below will continue to work, operating on the new array.

Sorry for the delay; I'll keep this PR open in a tab and check back in so we can get this merged.